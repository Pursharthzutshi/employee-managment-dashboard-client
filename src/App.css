html,
body {
    height: 100%;
    /* background: red; */
}


.dashboard-container {
    display: flex;
}

.dashboard-container {
    background: linear-gradient(to left,rgb(52, 195, 252),rgb(78, 184, 255));
    box-shadow: .2px .2px 22px .2px rgb(232, 232, 232);
    height: 100%;
}

.App {
    margin-top: 10px;
    /* height: 100%; */
    background: white;
    margin-left: 110px;
}

body::-webkit-scrollbar {
    background: linear-gradient(to left,rgb(255, 255, 255),rgb(255, 255, 255));
}

body::-webkit-scrollbar-thumb {
    background: linear-gradient(to left,rgb(245, 245, 245),rgb(235, 235, 235));
    border-radius: 2px;
}

body::-webkit-scrollbar-button {
    display: hidden;
}