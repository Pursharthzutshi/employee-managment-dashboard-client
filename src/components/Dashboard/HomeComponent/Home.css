/* .box:hover li{
    visibility: visible;
} */


/* Dashboard code */
.dashboard {
    padding: 2px 12px 2px 12px;
    margin-top: -7px;
    margin-left: 10px;
    /* width: 98.6%; */
    /* height: 100vh; */
    box-shadow: .2px .2px 22px .2px rgb(232, 232, 232);
    /* box-shadow: .2px .2px 22px .2px rgb(232, 232, 232); */
}

h3 {
    margin-top: 40px;
    margin-left: 20px;
}



.bar-chart-div,.pie-chart-div {
    border-radius: 8px;
    padding: 18px 18px 12px 18px;
    box-shadow: .2px .2px 2px .2px lightgray;
    margin-top: 40px;
    height: 18rem;
}

.bar-chart-div{
    width: 40%;

}

.pie-chart-div{
    display: flex;
}

.pie-chart-div{
    justify-content: center;
    align-items: center;
}
.pie-chart-div{
    width: 22%;
    margin-left: 10px;
}

.doughnut-chart {

    /* background: transparent; */
    /* width: 10%; */
    /* background: rgba(167, 167, 255, 0.038); */
    height: 200px;
}


.chart-div-container {
    display: flex;
    gap: 28px;
    margin-left: 12px;
}
/* 
.todo-list {
    box-shadow: .2px .2px 2px .2px lightgray;
    height: 40vh;
    width: 43%;
    overflow-y: scroll;
} */
