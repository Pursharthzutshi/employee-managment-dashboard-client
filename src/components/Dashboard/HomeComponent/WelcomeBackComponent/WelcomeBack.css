
.welcome-back-card-container,.welcome-back-card-image-message-div{
    display: flex;
}

.welcome-back-card-container{
    flex-direction:column;
}

.welcome-back-card-container{
    padding: 2px 2px 12px 12px;
    border-radius: 8px;
    /* background: linear-gradient(to left,rgb(248, 254, 255),rgb(244, 253, 253)); */
    margin-left: 10px;
    margin-top: 10px;
    width: 98.2%;
    /* height: 27vh; */
    box-shadow: .2px .2px 2px .2px lightgray;
}

.welcome-back-card-container p{
    line-height: 25px;
    font-size: .9rem;
    /* font-family: "Montserrat"; */
    padding: 2px 12px 2px 1px;
}

.welcome-back-card-image-message-div img{
    margin-top: 6px;
    width: 5rem;
    height: auto;
}

.welcome-back-button{
    font-size: .9rem;
    border: none;
    color:white;
    text-decoration: none;
    font-family: "Montserrat";
    padding: 2px 12px 2px 12px;
    background: linear-gradient(to left,rgb(0, 132, 214),rgb(0, 113, 183));
    
}