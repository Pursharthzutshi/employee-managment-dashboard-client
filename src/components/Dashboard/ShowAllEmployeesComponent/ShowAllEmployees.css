.show-all-employees-component {
    box-shadow: .2px .2px 22px .2px rgb(232, 232, 232);
    /* height: 100vh; */
    margin-left: 10px;
    margin-top: -10px;
    min-height: 93vh;
    padding: 2px 2px 32px 12px;
}

.employees-details-container {
    display: flex;
}

.employees-details-container {
    flex-direction: row;
    align-items: center;
}

.employees-details-container {
    flex-wrap: wrap;
    margin-top: 20px;

}

.search-employees-input {
    width: 50%;
    margin-left: 10px;
    border: none;
    letter-spacing: 1px;
    box-shadow: .2px .2px 2px .2px lightgray;
    padding: 6px 2px 6px 8px;
    font-family: "Montserrat";

}

.employees-details-div {
    display: flex;
    flex-direction: column;
    /* flex-wrap: wrap;
    text-overflow: ellipsis; */
    word-wrap: break-word;
    width: 15%;
    padding: 12px 22px 14px 22px;
    margin: 10px 2px 2px 10px;
    border-radius: 8px;
    border: 0px solid rgb(0, 0, 64);
    box-shadow: .2px .2px 2px .2px lightgray;
}

.employees-details-button{
    cursor: pointer;
    font-family: "Montserrat";
    color:white;
    border: none;
    white-space: nowrap;
    font-size: .77rem;
    width: 100%;
    border-radius: 4px;
    padding: 4px 12px 4px 11px;
    box-shadow: .2px .2px 12px .2px whitesmoke;
    background: linear-gradient(to left,rgb(0, 113, 212),rgb(0, 85, 170));
}

.employees-details-button:hover{
    background: white;
    transition: .2s;
    box-shadow: .2px .2px 2px .2px lightgray;    
    color:rgb(0, 113, 212);
}
/* 
{
    overflow: hidden;
    white-space: nowrap;

    align-content:; 
    overflow: hidden;
    inline-size: 150px;

    width: max-content;
     word-wrap: break-word;

} */


.no-employee-div{
    display: flex;
}

.no-employee-div{
    flex-direction: column;
}

.no-employee-div{
    height: 80vh;
    /* box-shadow: .2px .2px 12px .2px; */
    justify-content: center;
    align-items: center;
}

.add-new-employee-link{
    color:black;
    font-size: .9rem;
    /* border: 2px solid black; */
    text-decoration: none;
    padding:4px 12px 4px 12px;
    box-shadow: .2px .2px 2px .2px lightgray;
}