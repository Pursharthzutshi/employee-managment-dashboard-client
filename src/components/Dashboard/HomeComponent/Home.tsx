import "../HomeComponent/Home.css"
import { useEffect, useState } from "react";
// import Dropdown from 'react-dropdown';
import "../../../App.css"
import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    BarElement,
    Title,
    Tooltip,
    Legend,
    ArcElement

} from 'chart.js';

import NavBar from "../../NavBarComponent/NavBar";
import { gql, useQuery, useSubscription } from "@apollo/client";
import { useAppDispatch, useAppSelector } from "../../../ReduxHooks";
import { resetCounts, resetDepartmentCounts, setCount, setDepartmentCount, setGenderTypeCount } from "../../../ReduxSlicers/ChartsDetailsSlicer";
import { setCreateEmployeeNewAccountStatus } from "../../../ReduxSlicers/createEmployeeNewAccountStatusSlicer";
import EmployeeStatus from "./EmployeeStatusComponent/EmployeeStatus";
import CheckInStatus from "./EmployeeStatusComponent/CheckInStatusComponent/CheckInStatus";
import GenderTypeChart from "./HomeCharts/GenderTypeChart/GenderTypeChart";
import DepartmentChart from "./HomeCharts/DepartmentChart/DepartmentChart";
import { setDepartment } from "../../../ReduxSlicers/SignUpSlicer";
import CardsDetails from "./CardsDetailsComponent/CardsDetails";
import EmployeeOfTheMonth from "./EmployeeOfTheMonthComponent/EmployeeOfTheMonth";
import WelcomeBack from "./WelcomeBackComponent/WelcomeBack";

// Register the components
ChartJS.register(
    CategoryScale,
    LinearScale,
    BarElement,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend,
    ArcElement
);

const show_all_employees_data_query = gql`
query qd {
 showAllEmployee {
   genderType
   department
 }
}`

function Home() {


    const { data: employeesData, refetch } = useQuery(show_all_employees_data_query);

    const count = useAppSelector((state) => state.ChartsDetailsSlicer.count)
    const departmentCount = useAppSelector((state) => state.ChartsDetailsSlicer.departmentCount)
    const createEmployeeNewAccountStatus = useAppSelector((state) => state.createEmployeeNewAccountStatusSlicer.createEmployeeNewAccountStatus);

    const Dispatch = useAppDispatch()


    useEffect(() => {

        if (createEmployeeNewAccountStatus === true) {
            Dispatch(resetCounts());
            Dispatch(resetDepartmentCounts());
            Dispatch(setCreateEmployeeNewAccountStatus(false))
        }

        if (employeesData && employeesData.showAllEmployee) {

            if (count !== employeesData.showAllEmployee.length) {
                employeesData.showAllEmployee.map((employeesDataList: any) => {
                    return Dispatch(setGenderTypeCount(employeesDataList))
                })
                employeesData.showAllEmployee.map((employeesDataList: any) => {
                    return Dispatch(setDepartmentCount(employeesDataList))
                })
            }

            const totalDataCount = employeesData.showAllEmployee.length
            Dispatch(setCount(totalDataCount))

            refetch()
        }

    }, [employeesData])



    return (
        <div className="dashboard">


            <NavBar />

            <CheckInStatus />
            <WelcomeBack/>

            <CardsDetails />
    
            <div className="chart-div-container">

                <GenderTypeChart />
                <EmployeeOfTheMonth />
                <DepartmentChart />

            </div>
            <br></br>
            <div className="chart-div-container">


                {/* <EmployeeStatus /> */}
                {/* <DepartmentChart /> */}

            </div>

        </div>
    )
}

export default Home;

