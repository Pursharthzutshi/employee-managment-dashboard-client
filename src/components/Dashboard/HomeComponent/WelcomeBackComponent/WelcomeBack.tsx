import React from "react";
import image from "../../../RegisterComponent/images/employee.png"
import { useAppSelector } from "../../../../ReduxHooks";
import { gql } from "@apollo/client";
import "../WelcomeBackComponent/WelcomeBack.css"
import { Link } from "react-router-dom";

function WelcomeBack() {

    const savedLoggedInUserName = useAppSelector((state) => state.LocalStorageSlicer.savedLoggedInName)

    // const fetchAdminAccountDetails  = gql`

    // `

    return (
        <div className="welcome-back-card-container">
            <div className="welcome-back-card-image-message-div">
                <img src={image} />
                <h3>Welcome Back {savedLoggedInUserName}</h3>
            </div>
            <div className="welcome-back-card-text-button-div">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris lobortis accumsan elementum. Ut non fringilla tellus, vel iaculis orci. Vestibulum tristique finibus arcu id accumsan. Ut nec nisi vitae nulla posuere faucibus. Aliquam quis dui sit amet neque vestibulum lobortis. Curabitur lobortis nec augue ac euismod. Curabitur fermentum, tellus sed cursus ultrices, metus massa rutrum enim, at pretium mi lacus sed nulla. Sed sed ante risus.</p>
                <Link className="welcome-back-button" to="/">View Task</Link>
            </div>
        </div>

    )
}

export default WelcomeBack