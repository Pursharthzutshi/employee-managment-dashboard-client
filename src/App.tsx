import './App.css';
import LeftSidebar from './components/LeftSidebarComponent/LeftSidebar';
import { Route, Routes } from 'react-router-dom';
import EmployeesTaskManager from './components/Dashboard/EmployeesTaskManagerComponent/EmployeesTaskManager';
import Home from './components/Dashboard/HomeComponent/Home';
import ShowAllEmployees from './components/Dashboard/ShowAllEmployeesComponent/ShowAllEmployees';
import SignupAdmin from './components/RegisterComponent/SignUpComponent/CreateNewEmployeeAccount';
import { useEffect } from 'react';
import { useAppSelector } from './ReduxHooks';
import LoginUsers from './components/RegisterComponent/LoginComponent/LoginUsers';
import LoginAdmin from './components/RegisterComponent/LoginComponent/LoginAdmin';
import AssignedEmployeesTask from './components/Dashboard/UserComponents/AssignedEmployeesTask';
import EmployeesHome from './components/Dashboard/UserComponents/EmployeesHome';
import CreateNewEmployeeAccount from './components/Dashboard/CreateNewEmployeeAccountComponent/CreateNewEmployeeAccount';


function App() {


  const changeLoginForm = useAppSelector((state) => state.ChangeSignUpFormSlicer.changeLoginForm)

  const adminStatus = useAppSelector((state) => state.LocalStorageSlicer.adminStatus)

  const logOutButton = useAppSelector((state) => state.LocalStorageSlicer.showLogOutButtonElements)

  useEffect(() => {
    // console.log('adminStatus from localStorage:', localStorage.getItem('adminStatus'));
  }, [adminStatus])

  return (
    <div className="App">

      {/* <button onClick={() => Dispatch(setChangeComponent(true))}>change</button> */}

      {
        logOutButton ? <LeftSidebar /> : null
      }

      <Routes>
        <Route path="/home" element=
          {
            adminStatus ? <Home /> : <EmployeesHome />
          }
        />
        <Route path="/employeesTaskManagmentPage" element={

          adminStatus ? <EmployeesTaskManager /> : <AssignedEmployeesTask />
        }
        />

        <Route path="/showAllEmployeesData" element={<ShowAllEmployees />} />

        <Route path="/" element={
          changeLoginForm ? <LoginUsers /> : <LoginAdmin />}
        />

        <Route path="/createEmployeeNewAccount" element={<CreateNewEmployeeAccount />} />

        <Route path="/signUpAdmin" element={<SignupAdmin />} />

      </Routes>



    </div>
  );
}

export default App;
