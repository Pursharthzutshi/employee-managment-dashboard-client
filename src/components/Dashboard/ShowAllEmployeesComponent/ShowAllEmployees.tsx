import { useEffect, useState } from "react";
import { gql, useMutation, useQuery } from "@apollo/client";
import { useDispatch } from "react-redux";
import { setSearchFilter } from "../../../ReduxSlicers/SearchFilterSilcer";
import { useAppSelector } from "../../../ReduxHooks";
import { EmployeesAccountDataProps } from "../../../Types/ShowAllEmployeesComponentTypes";
import NavBar from "../../NavBarComponent/NavBar";


import "../ShowAllEmployeesComponent/ShowAllEmployees.css"
import { Link } from "react-router-dom";

const show_all_employees_data_query = gql`
query fetchemployeesDataQuery{
   showAllEmployee {
   uid
   name
   emailId
}
}`

const update_Employee_Of_The_Month_query = gql`
mutation updateEmployeeOfTheMonth($updateEmployeeOfTheMonthParameters: updateEmployeeOfTheMonthInput!){
  updateEmployeeOfTheMonth(updateEmployeeOfTheMonthParameters: $updateEmployeeOfTheMonthParameters) {
    uid,
    employeeOfTheMonth
  }
}
  `

function ShowAllEmployees() {
    const searchFilter = useAppSelector((state) => state.SearchFilterSilcer.SearchFilter)
    const [totalEmployeeDetailsCount,setTotalEmployeeDetailsCount] = useState(0);
   
    const createEmployeeNewAccountStatus = useAppSelector((state)=>state.createEmployeeNewAccountStatusSlicer.createEmployeeNewAccountStatus)

    const Dispatch = useDispatch()

    const [assignEmployeeOfTheMonth] = useMutation(update_Employee_Of_The_Month_query,({
        onError:(err)=>{
            console.log(err)
        }
    }));


    const { data: ShowAllEmployeesData, loading,refetch} = useQuery(show_all_employees_data_query, {
        onCompleted: (data) => {
            console.log(data)
            refetch()
        }
    }
    );

    // const assignEmployeeOfTheMonth = (val:any)=>{
    //     console.log(val)
    // }

    const adminStatus = useAppSelector((state) => state.LocalStorageSlicer.adminStatus)

    if (loading) return <p>Loading...</p>

    return (
        <div className="show-all-employees-component">

            <NavBar />
            {/* <input  type="search" /> */}

  
    <div>
    <h3>All Employees</h3>
    <input onChange={(e) => Dispatch(setSearchFilter(e.target.value))} className="search-employees-input" placeholder="Search Employees" type="text" />
    <div className="employees-details-container">
        { 
            ShowAllEmployeesData.showAllEmployee.filter((filteredEmployeesAccountData: EmployeesAccountDataProps) => {

                if (filteredEmployeesAccountData.name.toLowerCase().includes(searchFilter.toLowerCase())) {
                    console.log(filteredEmployeesAccountData)
                    return filteredEmployeesAccountData;
                } else if (searchFilter === "") {
                    return filteredEmployeesAccountData;
                }
            }).map((EmployeesAccountData: EmployeesAccountDataProps) => {
                console.log(EmployeesAccountData)
                return (
                    <div className="employees-details-div" >
                        <strong>Name:</strong><p>{EmployeesAccountData.name}</p>
                        <strong>Email ID:</strong><p className="email-id">{EmployeesAccountData.emailId}</p>
                        {adminStatus ? <button onClick={() => {
                            assignEmployeeOfTheMonth({
                                variables: {
                                    updateEmployeeOfTheMonthParameters: {
                                        uid: EmployeesAccountData.uid,
                                        employeeOfTheMonth: true
                                    },
                                },
                            })

                        }} className="employees-details-button">Assign Employee of the month</button> : null}
                    </div>
                )
            })
        }
    </div>
    </div>
    {/* :
    <div className="no-employee-div">
        <h3>No Employees </h3>
        <p>Please Add New Employee </p>
        <Link className="add-new-employee-link" to="/createEmployeeNewAccount">Add New Employees</Link>
    </div> */}

         
        </div>
    )
}

export default ShowAllEmployees;