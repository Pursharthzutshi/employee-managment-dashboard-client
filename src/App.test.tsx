import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import App from './App';
import {act} from 'react';
import { Provider } from 'react-redux';
import ShowAllEmployees from './components/Dashboard/ShowAllEmployeesComponent/ShowAllEmployees';
import { store } from '../src/ReduxStore/store';
import Home from './components/Dashboard/HomeComponent/Home';
import WelcomeBack from './components/Dashboard/HomeComponent/WelcomeBackComponent/WelcomeBack';
import userEvent from '@testing-library/user-event';
import { ApolloProvider } from '@apollo/client';
import { client } from '.';


it("snapshot",async ()=>{
    
      const {container} = await act(async () => render(
          <Provider store={store}>
            <WelcomeBack/>
          </Provider>
  ))
      expect(container.querySelector("p")).toMatchSnapshot();
  
  })
  

  it("check",async()=>{
    
    const {container} = render(
      <ApolloProvider client={client}>
      <Provider store ={store}>
      <ShowAllEmployees/>
      </Provider>
      </ApolloProvider>,
  )

    const input = screen.getByTestId("search-input")
    userEvent.type(input,"")

    await waitFor(()=>{
      expect(container.querySelectorAll(".employee-name").length).toBe(0)
    })
  })