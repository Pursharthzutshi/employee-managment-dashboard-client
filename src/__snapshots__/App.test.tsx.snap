// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<p
  class="font-bold text-lg ml-3 mt-5"
>
  Welcome Back 
</p>
`;
